server/src/main/java/org/candlepin/resource/DeletedConsumerResource.java

```diff
@@ -51,17 +40,11 @@ public DeletedConsumerResource(DeletedConsumerCurator deletedConsumerCurator,
         this.translator = translator;
     }
 
-    @ApiOperation(
-        notes = "Retrieves a list of Deleted Consumers By deletion date or all. " +
-        "List returned is the deleted Consumers.",
-        value = "listByDate", response = DeletedConsumerDTO.class, responseContainer = "list")
-    @ApiResponses({ @ApiResponse(code = 400, message = ""), @ApiResponse(code = 404, message = "") })
-    @GET
-    @Produces(MediaType.APPLICATION_JSON)
-    public CandlepinQuery<DeletedConsumerDTO> listByDate(@QueryParam("date") String dateStr) {
-        return this.translator.translateQuery(dateStr != null ?
+    @Override
+    public Stream<DeletedConsumerDTO> listByDate(String dateStr) {
+        CandlepinQuery<DeletedConsumerDTO> query = this.translator.translateQuery(dateStr != null ?
             this.deletedConsumerCurator.findByDate(ResourceDateParser.parseDateString(dateStr)) :
-            this.deletedConsumerCurator.listAll(),
-            DeletedConsumerDTO.class);
+            this.deletedConsumerCurator.listAll(), DeletedConsumerDTO.class);
+        return StreamSupport.stream(query.spliterator(), false);
```

Comment:
This doesn't actually work in all cases. Since the processing of the stream occurs later, the connection backing the CPQ can close and it'll trigger an exception instead of returning the result.

For now, just return the CandlepinQuery object as normal, and set the container type to "Iterable" in the yaml.

Reply:
@Ceiu 
I had tested this API locally and it was working. For now, I have updated code as per your suggestion i.e. use Iterable instead of Stream. 
 
If we use Stream then would it fail in a larger response? Could you please share the scenarios in which this will throw an exception in the case of Stream.

Reply:
The easier thing to do would be to take a look at CandlepinQueryInterceptor.

There are two risks to dropping the interceptor and just using a stream directly:
1) We give up our control over when the stream is processed vs when the connection backing it is closed. We've had issues in the past with race conditions where the connection would close and cause various exceptions when the stream is processed. I would have to dip into the git history to give you concrete examples.
2) Nested objects straight up fail in many cases. This was the driving feature here that caused the CandlepinQueryInterceptor to start using its own connection.

I think the long-term goal is to remove CPQ entirely and start using properly designed streams, but that is not something we're willing/able to address in these PRs. For the immediate term, we can move into the spec-first world by setting the container type to be Iterable, but then returning the CandlepinQuery as-is and letting the interceptor pick it up and process it safely. We can even leverage Java's covariant return typing to let the generated API interface define it as an Iterable, but our class define it as a CandlepinQuery as it does today.

Reply:
Thanks @Ceiu for details.